### ISSUE TITLE
Support for multiple sample rate fragments

### ISSUE THREAD

**Issue by giuliomoro** [BrechtDeMan/WebAudioEvaluationTool#101]
Inspired by #100  , I re-up this soundsoftware issue by Brecht (and a me in disguise ??).

@BrechtDeMan said:
It was suggested by a user _[edit: I think it was me]_ of the tool that instead of enforcing a specific sample rate to avoid resampling, multiple sample rates could be allowed where several file names or folders could point to files in different sample rates. 
For instance, if the detected sample rate is 44.1k, the 'trumpet-44.wav' sample is loaded, if the detected sample rate is 96k 'trumpet-96.wav' is loaded, and otherwise an error message is shown (and/or audio is resampled?).

@nickjillings said:
Not a bad idea, but then each file would need its sample rate explicitly stated before download to make a decision. In specification it should not be too hard since it should have an alternative. This would be a good 'webby' thing to have as well if perhaps there was a redundant server system, giving a 'fallback' file should the first fail to load.

**Comment by nickjillings** [comment]
Added two new items to the `<audioelement>` node.

Added an attribute `sampleRate` which must be greater than or equal to 1. Mesaured in Hz.

Also added child nodes to the `<audioelement>` node called `<alternative>`. These take only two attributes, a url (mandatory) to indicate an alternative resource location for the file, and sampleRate to indicate the same file encoded in a different audio sampleRate.

**Comment by nickjillings** [comment]
Tested, it now loads up an alternative URL if the primary fails (returns error code). Therefore if a server returns an error (such as 404 because of deletions, or 500 from server overloads) it will try to load from the alternative locations.

**Comment by nickjillings** [comment]
Implemented in full in Dev_main (dc26aab)

On creation of a buffer, if will pull to the head of queue thos which match the reported system rate. The rest are then stacked in order.

Suppose we have:
```xml
<audioelement id="track-1" url="media/audio.wav">
    <alternative sampleRate="44100" url="media/44k.wav" />
    <alternative sampleRate="48000" url="media/48k.wav" />
</audioelement>
```
If this loads on a system with `sampleRate == 44100` then the first order is: `["media/44k.wav", "media/audio.wav", "media48k.wav"]`

If on a system with `sampleRate == 48000` then the order is `["media/48k.wav", "media/audio.wav", "media44k.wav"]`

On a system with any other sample rate it is `["media/audio.wav", "media/44k.wav", "media48k.wav"]`

Then the loading process resumes where if the first URL doesn't work, it loads the next in list. If multiple alternatives are given with the same sampleRate, they are placed at the head in order. So it is possible to generate a full main, backup for each sample rate:

```xml
<audioelement id="track-1" url="media/defaul.wav">
    <alternative sampleRate="44100" url="media/44k.wav" />
    <alternative sampleRate="44100" url="media/44kB.wav" />
    <alternative sampleRate="48000" url="media/48k.wav" />
    <alternative sampleRate="48000" url="media/48kB.wav" />
    <alternative sampleRate="96000" url="media/96k.wav" />
    <alternative sampleRate="96000" url="media/96kB.wav" />
</audioelement>
```

**Comment by nickjillings** [comment]
The alternatives must be the **Fully Qualified Name**. It will not append the hostURL attribute to these. So in reality the above example should be:

```xml
<audioelement id="track-1" url="media/audio.wav">
    <alternative sampleRate="44100" url="http://my.other.server/media/44k.wav" />
    <alternative sampleRate="48000" url="http://my.second.server/media/48k.wav" />
</audioelement>
```
